### PR TITLE
[Optim][Rprop] Replace new().resize_as_() by torch.full_like()

### PR DESCRIPTION
**Pull request by wz337**
As titled.

### CODE REVIEW

File: torch/optim/rprop.py [hunk 1]
```diff
@@ -65,13 +65,12 @@ def _init_group(self, group, params, grads, prevs, step_sizes):
...
                     state["step_size"] = (
-                        grad.new()
-                        .resize_as_(grad)
+                        torch.empty_like(grad)
                         .fill_(complex(group["lr"], group["lr"]))
                     )
                 else:
                     state["step_size"] = (
-                        grad.new().resize_as_(grad).fill_(group["lr"])
+                        torch.empty_like(grad).fill_(group["lr"])
```
Comment (vadimkantorov) [inline torch/optim/rprop.py:73]: could this instead be `torch.full_like(grad, group["lr"])`?

Comment (wz337) [reply torch/optim/rprop.py]: @vadimkantorov Thanks for the suggestion! 

@janeyx99, do you know the answer to @vadimkantorov 's 2nd question?

**Comment by wz337** [comment]
@janeyx99 Seems CI is working fine. Should we proceed with the changes?

File: torch/optim/rprop.py (hunk 1 above)
Comment (janeyx99) [reply torch/optim/rprop.py]: Currently, it is the same shape as param/grad

**Review by janeyx99 — APPROVED** [review]
This seems pleasant to me!

Digging into the code reveals that calling `.new()` here preserves the layout, device, and dtype, and calling `.resize_as_` will also propagate the shape. `full_like` seems to propagate more with `empty_like` so we should be safe here.

**Comment by vadimkantorov** [comment]
@janeyx99 Is here lr/step_size per parameter element? E.g. same shape as the param/grad?

**Comment by janeyx99** [comment]
> @janeyx99 Is here lr/step_size per parameter element? E.g. same shape as the param/grad?

yes, it is currently per element (vs per param) and the same shape as the param/grad. In the code, the elements will vary based on sign of grad * prev_grad element-wise.

**Pull request closed without merging**